models/demos/wormhole/mnist/tests/test_perf_mnist_wh.py

```diff
@@ -0,0 +1,129 @@
+
+transform = transforms.Compose([transforms.ToTensor()])
+test_dataset = datasets.MNIST(root="./data", train=False, transform=None, download=True)
+
+
+@skip_for_grayskull()
+def get_expected_times(tt_mnist):
+    if is_wormhole_b0():
+        return {
+            tt_mnist: (10.89, 0.017),
+        }[tt_mnist]
+
+
+@pytest.mark.models_performance_bare_metal
+@pytest.mark.models_performance_virtual_machine
+@pytest.mark.parametrize(
+    "batch_size",
+    [256],
+)
+@pytest.mark.parametrize(
+    "tt_mnist",
+    [tt_mnist],
+)
+def test_performance_mnist(mesh_device, batch_size, tt_mnist, model_location_generator):
+    state_dict = torch.load(model_location_generator("mnist_model.pt", model_subdir="mnist"))
+    model = MnistModel(state_dict)
+    model = model.eval()
+    disable_persistent_kernel_cache()
+    transform = transforms.Compose([transforms.ToTensor()])
+    test_dataset = datasets.MNIST(root="./data", train=False, transform=transform, download=True)
+    dataloader = DataLoader(test_dataset, batch_size=batch_size)
+    x, labels = next(iter(dataloader))
+    mesh_device_flag = is_wormhole_b0() and ttnn.GetNumAvailableDevices() == 2
+    batch_size = batch_size if mesh_device_flag else batch_size // 2
+    inputs_mesh_mapper = ttnn.ShardTensorToMesh(mesh_device, dim=0)
+    weights_mesh_mapper = ttnn.ReplicateTensorToMesh(mesh_device)
+    output_mesh_composer = ttnn.ConcatMeshToTensor(mesh_device, dim=0)
+    with ttnn.distribute(ttnn.ReplicateTensorToMesh(mesh_device)):
+        parameters = preprocess_model_parameters(
+            initialize_model=lambda: model,
+            device=mesh_device,
+        )
+
+    x = ttnn.from_torch(x, dtype=ttnn.bfloat16, mesh_mapper=inputs_mesh_mapper, device=mesh_device)
+    durations = []
+
+    for _ in range(2):
+        start = time.time()
+
+        ttnn_output = tt_mnist.mnist(mesh_device, batch_size, x, parameters)
+        end = time.time()
+        durations.append(end - start)
+        enable_persistent_kernel_cache()
+
+    inference_and_compile_time, *inference_times = durations
+    inference_time = sum(inference_times) / len(inference_times)
+    expected_compile_time, expected_inference_time = get_expected_times(tt_mnist)
+
+    prep_perf_report(
+        model_name="MNIST",
+        batch_size=batch_size,
+        inference_and_compile_time=inference_and_compile_time,
+        inference_time=inference_time,
+        expected_compile_time=expected_compile_time,
+        expected_inference_time=expected_inference_time,
+        comments="",
+        inference_time_cpu=0.0,
+    )
+
+    logger.info(f"Compile time: {inference_and_compile_time - inference_time}")
+    logger.info(f"Inference time: {inference_time}")
+    logger.info(f"Inference times: {inference_times}")
+    logger.info(f"Sample(s) per second: {1 / inference_time * batch_size}")
+    assert (
+        inference_time < expected_inference_time
```

Comment:
like the single device PR, please make sure the expected inference time is close to actual inference time we are getting in CI.

Reply:
Updated the PR  and triggered CIs